src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java

```diff
@@ -1,28 +1,170 @@
 package org.opentripplanner.middleware.controllers.api;
 
 import com.amazonaws.services.apigateway.model.CreateApiKeyResult;
+import com.beerboy.ss.ApiEndpoint;
 import org.eclipse.jetty.http.HttpStatus;
+import org.opentripplanner.middleware.auth.Auth0Connection;
 import org.opentripplanner.middleware.auth.Auth0UserProfile;
-import org.opentripplanner.middleware.utils.ApiGatewayUtils;
+import org.opentripplanner.middleware.models.ApiKey;
 import org.opentripplanner.middleware.models.ApiUser;
 import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.utils.ApiGatewayUtils;
+import org.opentripplanner.middleware.utils.HttpUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
 import spark.HaltException;
 import spark.Request;
+import spark.Response;
+
+import java.util.Collections;
 
+import static com.beerboy.ss.descriptor.MethodDescriptor.path;
 import static org.opentripplanner.middleware.OtpMiddlewareMain.getConfigPropertyAsText;
+import static org.opentripplanner.middleware.auth.Auth0Connection.isUserAdmin;
 import static org.opentripplanner.middleware.utils.JsonUtils.logMessageAndHalt;
 
 /**
- * Implementation of the {@link AbstractUserController} for {@link ApiUser}.
+ * Implementation of the {@link AbstractUserController} for {@link ApiUser}. This controller also contains methods for
+ * managing an {@link ApiUser}'s API keys.
  */
 public class ApiUserController extends AbstractUserController<ApiUser> {
-    private static final String DEFAULT_USAGE_PLAN_ID
-        = getConfigPropertyAsText("DEFAULT_USAGE_PLAN_ID");
+    private static final String DEFAULT_USAGE_PLAN_ID = getConfigPropertyAsText("DEFAULT_USAGE_PLAN_ID");
+    private static final String API_KEY_PATH = "/apikey";
+    private static final int API_KEY_LIMIT_PER_USER = 2;
+    private static final String API_KEY_ID_PARAM = "/:apiKeyId";
 
     public ApiUserController(String apiPrefix) {
         super(apiPrefix, Persistence.apiUsers, "secure/application");
     }
 
+    @Override
+    protected void buildEndpoint(ApiEndpoint baseEndpoint) {
+        LOG.info("Registering path {}.", ROOT_ROUTE + API_KEY_PATH);
+
+        // Add the api key route BEFORE the regular CRUD methods
+        // TODO: Define usage plan id as an optional parameter under Spark Swagger. 'withParam' method does not register
+        //  the parameter and 'withPathParam' is not optional (even with 'withRequired(false))'. For the time being
+        //  appending '?usagePlanId=<id>' works.
```

Comment:
I think this TODO should be closer to the Create API Key method to avoid confusion.

Reply:
Updated.